src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIModelMapper.ChatMessage.cs

```diff
@@ -0,0 +1,251 @@
+                                callRequest.Name,
+                                new(JsonSerializer.SerializeToUtf8Bytes(
+                                    callRequest.Arguments,
+                                    options.GetTypeInfo(typeof(IDictionary<string, object?>))))));
+                    }
+                }
+
+                if (input.AdditionalProperties?.TryGetValue(nameof(message.Refusal), out string? refusal) is true)
+                {
+                    message.Refusal = refusal;
+                }
+
+                yield return message;
+            }
+        }
+    }
+
+    private static List<AIContent> FromOpenAIChatContent(IList<ChatMessageContentPart> openAiMessageContentParts)
+    {
+        List<AIContent> contents = new();
+        foreach (var openAiContentPart in openAiMessageContentParts)
+        {
+            switch (openAiContentPart.Kind)
+            {
+                case ChatMessageContentPartKind.Text:
+                    contents.Add(new TextContent(openAiContentPart.Text));
+                    break;
+
+                case ChatMessageContentPartKind.Image when (openAiContentPart.ImageBytes is { } bytes):
+                    contents.Add(new ImageContent(bytes.ToArray(), openAiContentPart.ImageBytesMediaType));
+                    break;
+
+                case ChatMessageContentPartKind.Image:
+                    contents.Add(new ImageContent(openAiContentPart.ImageUri?.ToString() ?? string.Empty));
+                    break;
+
+            }
+        }
+
+        return contents;
+    }
+
+    /// <summary>Converts a list of <see cref="AIContent"/> to a list of <see cref="ChatMessageContentPart"/>.</summary>
+    private static List<ChatMessageContentPart> ToOpenAIChatContent(IList<AIContent> contents)
+    {
+        List<ChatMessageContentPart> parts = [];
+        foreach (var content in contents)
+        {
+            switch (content)
+            {
+                case TextContent textContent:
+                    parts.Add(ChatMessageContentPart.CreateTextPart(textContent.Text));
+                    break;
+
+                case ImageContent imageContent when imageContent.Data is { IsEmpty: false } data:
+                    parts.Add(ChatMessageContentPart.CreateImagePart(BinaryData.FromBytes(data), imageContent.MediaType));
+                    break;
+
+                case ImageContent imageContent when imageContent.Uri is string uri:
+                    parts.Add(ChatMessageContentPart.CreateImagePart(new Uri(uri)));
+                    break;
+            }
+        }
+
+        if (parts.Count == 0)
+        {
+            parts.Add(ChatMessageContentPart.CreateTextPart(string.Empty));
+        }
+
+        return parts;
+    }
+
+#pragma warning disable S3011 // Reflection should not be used to increase accessibility of classes, methods, or fields
+    private static readonly Func<ChatCompletionOptions, IList<OpenAI.Chat.ChatMessage>> _getMessagesAccessor =
```

Comment:
Can we ensure there's an open issue in the OpenAI repo about exposing the ability to access this information?